nomad/node_endpoint.go

```diff
@@ -434,6 +434,62 @@ func (n *Node) UpdateDrain(args *structs.NodeUpdateDrainRequest,
 	return nil
 }
 
+// UpdateFreeze is used to update the freeze mode of a client node
+func (n *Node) UpdateFreeze(args *structs.NodeUpdateFreezeRequest,
+	reply *structs.NodeFreezeUpdateResponse) error {
+	if done, err := n.srv.forward("Node.UpdateFreeze", args, args, reply); done {
+		return err
+	}
+	defer metrics.MeasureSince([]string{"nomad", "client", "update_freeze"}, time.Now())
+
+	// Verify the arguments
+	if args.NodeID == "" {
+		return fmt.Errorf("missing node ID for freeze update")
+	}
+
+	// Look for the node
+	snap, err := n.srv.fsm.State().Snapshot()
+	if err != nil {
+		return err
+	}
+	ws := memdb.NewWatchSet()
+	node, err := snap.NodeByID(ws, args.NodeID)
+	if err != nil {
+		return err
+	}
+	if node == nil {
+		return fmt.Errorf("node not found")
+	}
+
+	// Update the timestamp to
+	node.StatusUpdatedAt = time.Now().Unix()
+
+	// Commit this update via Raft
+	var index uint64
+	if node.Freeze != args.Freeze {
+		_, index, err = n.srv.raftApply(structs.NodeUpdateFreezeRequestType, args)
+		if err != nil {
+			n.srv.logger.Printf("[ERR] nomad.client: freeze update failed: %v", err)
+			return err
+		}
+		reply.NodeModifyIndex = index
+	}
+
+	// Always attempt to create Node evaluations because there may be a System
```

Comment:
do we need this for freeze? i guess no evaluation should happen against the client at all? since they would be NOOP anyway for frozen clients

Reply:
but what to do with system jobs. Drained node can run this type of job, so to be consistent `node-freeze` should also support this.